File: Online Judges/LeetCode/June LeetCoding Challenge/Week-2/Cheapest Flights Within K Stops.cpp
class Solution
{
public:
    int findCheapestPrice(int n, vector<vector<int>> &flights, int src, int dst, int k)
    {
        int cost[101][101];
        for (int i = 0; i <= n; i++)
        {
            for (int j = 0; j <= n; j++)
            {
                cost[i][j] = 1e8;
            }
        }
        cost[0][src] = 0;
        for (int i = 1; i <= k + 1; i++)
        {
            cost[i][src] = 0;
            for (auto flight : flights)
            {
                cost[i][flight[1]] = min(cost[i][flight[1]], cost[i - 1][flight[0]] + flight[2]);
            }
        }
        return cost[k + 1][dst] >= 1e8 ? -1 : cost[k + 1][dst];
    }
};

File: Online Judges/LeetCode/June LeetCoding Challenge/Week-2/Cheapest Flights Within K Stops.py
class Solution:
    def findCheapestPrice(self, n: int, flights: List[List[int]], src: int, dst: int, K: int) -> int:
        cost = [[1e8]*(n) for j in range(K + 2)]

        cost[0][src] = 0

        for i in range(1, K+2):
            cost[i][src] = 0
            for u, v, k in flights:
                cost[i][v] = min(cost[i][v], cost[i-1][u] + k)

        return cost[K+1][dst] if cost[K+1][dst] != 1e8 else -1
